pkg/cloud/scope/clients.go

```diff
@@ -17,16 +17,74 @@ limitations under the License.
 package scope
 
 import (
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/aws/aws-sdk-go/service/elb"
 	"github.com/aws/aws-sdk-go/service/elb/elbiface"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
 	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi/resourcegroupstaggingapiiface"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
 	"github.com/aws/aws-sdk-go/service/secretsmanager/secretsmanageriface"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/cloud"
+	"sigs.k8s.io/cluster-api-provider-aws/pkg/record"
+	"sigs.k8s.io/cluster-api-provider-aws/version"
 )
 
-// AWSClients contains all the aws clients used by the scopes.
-type AWSClients struct {
-	EC2             ec2iface.EC2API
-	ELB             elbiface.ELBAPI
-	SecretsManager  secretsmanageriface.SecretsManagerAPI
-	ResourceTagging resourcegroupstaggingapiiface.ResourceGroupsTaggingAPIAPI
+// NewEC2Client creates a new EC2 API client for a given session
+func NewEC2Client(session cloud.Session, target runtime.Object) ec2iface.EC2API {
+	ec2Client := ec2.New(session.Session())
+	ec2Client.Handlers.Build.PushFrontNamed(getUserAgentHandler())
+	ec2Client.Handlers.Complete.PushBack(recordAWSPermissionsIssue(target))
+
+	return ec2Client
+}
+
+// NewELBClient creates a new ELB API client for a given session
+func NewELBClient(session cloud.Session, target runtime.Object) elbiface.ELBAPI {
+	elbClient := elb.New(session.Session())
+	elbClient.Handlers.Build.PushFrontNamed(getUserAgentHandler())
+	elbClient.Handlers.Complete.PushBack(recordAWSPermissionsIssue(target))
+
+	return elbClient
+}
+
+// NewResourgeTaggingClient creates a new Resource Tagging API client for a given session
+func NewResourgeTaggingClient(session cloud.Session, target runtime.Object) resourcegroupstaggingapiiface.ResourceGroupsTaggingAPIAPI {
+	resourceTagging := resourcegroupstaggingapi.New(session.Session())
+	resourceTagging.Handlers.Build.PushFrontNamed(getUserAgentHandler())
+	resourceTagging.Handlers.Complete.PushBack(recordAWSPermissionsIssue(target))
+
+	return resourceTagging
+}
+
+// NewSecretsManagerClient creates a new Secrets API client for a given session
+func NewSecretsManagerClient(session cloud.Session, target runtime.Object) secretsmanageriface.SecretsManagerAPI {
+	sClient := secretsmanager.New(session.Session())
+	sClient.Handlers.Build.PushFrontNamed(getUserAgentHandler())
```

Comment:
Review the rebasing again, as it looks like the metrics collection from https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1807 has got wiped out here.

Reply:
@randomvariable - ah yes, dodgey rebase by me. Sorry about that, metrics re-added to client creation.